File: src/SkyBlock/enchants/armor/Frozen.php
<?php


namespace SkyBlock\enchants\armor;


use pocketmine\entity\effect\EffectInstance;
use pocketmine\entity\effect\VanillaEffects;
use pocketmine\player\Player;

class Frozen extends BaseArmorEnchant {

    static int $id = 126;


    public function getLevel(int $level = 1) : int {
        if ($level > 12) {
            return 3;
        } else if ($level > 6) {
            return 2;
        } else {
            return 1;
        }
    }

    /**
     * @param Player $attacker
     * @param Player $victim
     * @param int    $enchantmentLevel
     */
    public function onActivation(Player $attacker, Player $victim, int $enchantmentLevel) : void {
        $this->sendActivation($victim, "§bFrozen §aActivated!");
        $this->sendActivation($attacker, "§cStruck by §bFrozen §cEnchant!");
        $effect = new EffectInstance(VanillaEffects::SLOWNESS(), 20 * 3, $this->getLevel($enchantmentLevel), false);
        $attacker->getEffects()->add($effect);
        $effect = new EffectInstance(VanillaEffects::BLINDNESS(), $this->getDuration($enchantmentLevel), 1, false);
        $attacker->getEffects()->add($effect);
    }

}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/armor/Frozen.java
package dev.skyblock.enchants.armor;

import dev.skyblock.enchants.BaseArmorEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;
import java.util.concurrent.ThreadLocalRandom;

public class Frozen extends BaseArmorEnchant {
    public static final int ID = 126;

    public Frozen() {
        super(ID, "Frozen", 15);
    }

    @Override
    public boolean isApplicableTo(Player player, int level) {
        return ThreadLocalRandom.current().nextInt(1, 26) == 1;
    }

    @Override
    public void onActivation(Player victim, EntityDamageByEntityEvent ev, int level) {
        if (ev.getDamager() instanceof Player attacker) {
            victim.sendMessage("§bFrozen §aActivated!");
            attacker.sendMessage("§cStruck by §bFrozen §cEnchant!");

            int amplifier = level > 12 ? 2 : (level > 6 ? 1 : 0);
            attacker.addPotionEffect(new PotionEffect(PotionEffectType.SLOWNESS, 20 * 3, amplifier));
            attacker.addPotionEffect(new PotionEffect(PotionEffectType.BLINDNESS, (int) (Math.ceil(level / 2.0) * 20), 1));
        }
    }
}
